internal/service/vmservice/vm.go

```diff
@@ -311,10 +315,15 @@ func getMachineAddresses(scope *scope.MachineScope) ([]clusterv1.MachineAddress,
 }
 
 func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneResponse, error) {
+	vmid, err := getVMID(ctx, scope)
+	if err != nil {
+		return proxmox.VMCloneResponse{}, err
```

Comment:
there's only one remaining thing:
you will probably need to fail the machine provisioning here.

```go
if err != nil {
  if errors.Is(err, ErrNoVMIDInRangeFree) {
    // fail the machine, by setting failureReason, and failureMessage
  }
}

```